src/servers/http_server.cc

```diff
@@ -2325,13 +2342,10 @@ HTTPAPIServer::HandleInfer(
 
   if (err != nullptr) {
     LOG_VERBOSE(1) << "Infer failed: " << TRITONSERVER_ErrorMessage(err);
-    EVBufferAddErrorJson(req->buffer_out, err);
-    evhtp_send_reply(req, EVHTP_RES_BADREQ);
+    HandleInferErrors(req, err);
```

Comment:
Control reaches here only if there is error before the model executes the request, so also need the check after `TRITONSERVER_InferenceResponseError()` where the error during execution will be reported to